### DISCUSSION TITLE
Need Support for SDRplay RSP 2

### DISCUSSION THREAD

**Discussion by pmaine** [AlexandreRouma/SDRPlusPlus#285] [Q&A] [answered]
How can I enable a SDRplay RSP 2 in SDR++?

**Comment by AlexandreRouma** [comment, marked as answer]
all SDRplay devices are supported. Make sure you've got the SDRplay API installed and started.

**Reply by pmaine** [reply]
I went to the SDRplay website to get the API and documentation.
thank you

**Reply by pmaine** [reply]
Github wont let me mark as answered

**Reply by pmaine** [reply]
I downloaded and installed the SDRPlay API. You can see it running here:

pi@raspberrypi:~ $ ps aux | grep sdrplay
root       537  0.5  0.0  13856  1216 ?        Ssl  14:34   0:05 /usr/local/bin/sdrplay_apiService
pi       11249  0.0  0.0   7348   536 pts/0    R+   14:51   0:00 grep --color=auto sdrplay

After I refresh I still don't see a SDRPlay as a Source option.

Do I need to build the software again since I now have the API?

Can I sudo make install?

**Reply by AlexandreRouma** [reply]
make sure to restart your computer

**Reply by pmaine** [reply]
I restarted my computer.
I next ran the following command:
pi@raspberrypi:~ $ ps aux | grep sdrplay

root       508  0.3  0.0  13856  1208 ?        Ssl  17:50   0:00 /usr/local/bin/sdrplay_apiService
pi        2691  0.0  0.0   7348   580 pts/0    S+   17:53   0:00 grep --color=auto sdrplay

Next, I started SDR++ and I still do not have a source for SDRPlay
![NoSourceSDRPlay](https://user-images.githubusercontent.com/5252230/130301086-268da904-e410-4d2b-b4de-59238141d47c.png)

**Reply by pmaine** [reply]
In the module manager I don't see a source for SDRPlay.

**Reply by AlexandreRouma** [reply]
I see it's a pi, did you have the SDRplay module enabled when building? (see readme about enabled and disabled modules)

**Reply by pmaine** [reply]
Hi Alexandre
I really believe in the software you are developing. I posted a very positive SDR++ post on Reddit as u/Snoo-76541. That said the documentation is not clear to me. Since you wrote the software I'm sure you think the directions are crystal clear but to me there is much room for improvement. Rather than complain I want to help.

Who am I - I am a lead software developer for a major corporation with more than 30 years experience as a developer. I mostly develop in C# .NET and PHP these days. In the earlier years of my career I wrote extensively in c and c++ - but that was many years ago...

I am trying to build for a raspberry pi and I want to take my learnings and evangelize your software to many Ham radio enthusiasts around the world. The other Linux SDR software packages for Linux are very poor when compared to yours.

I am not building a DEV environment at this time. I do not see a JSON config file.

Please advice.
Highest regards,
Paul KI5MIV

**Reply by AlexandreRouma** [reply]
Hello,

You shouldn't have to modify any config file.

the important part of the readme is [this table](https://github.com/AlexandreRouma/SDRPlusPlus#module-list),
As you can see the `sdrplay_source` module is listed as not being built by default. This is because the sdrplay API is quite heavy and I 
wouldn't want to forced users to install it before compiling with default settings. The tables gives you the cmake parameter to use to enable building the module `OPT_BUILD_SDRPLAY_SOURCE`.

Since I'm not sure what state your build environment is, I'd recommend first uninstall SDR++ (sudo make uninstall), then delete  your SDR++ source, and reclone. Then I recommend also deleting the SDR++ settings just in case `rm -r ~/.config/sdrpp`

Then, apply the patch mentioned in a previous thread to allow SDR++ to work on the RPi 4.

Now lets get to the build step. CD to the directory of SDR++ and run the following commands (Notice the -DOPT_BUILD_SDRPLAY_SOURCE=ON cmake parameter enabling the sdrplay source module):

```sh
mkdir build
cd build
cmake .. -DOPT_BUILD_SDRPLAY_SOURCE=ON
make
sudo make install
```

That is all you have to do.

By the way, on a Pi 4B, I doubt it'll run any faster than 3 or 4MS/s

**Reply by pmaine** [reply]
That worked!!!
I am using A pi4B 8GB with an argon M2 enclosure. I over clocked the processor to 2Ghz.

I was able to sample at 6Mhz successfully.
The audio started distorting when I reached 7Mhz.

**Reply by pmaine** [reply]
I also use a SATA SSD that's 256GB
